sdk/src/main/java/io/opentdf/platform/sdk/nanotdf/ResourceLocator.java

```diff
@@ -73,17 +147,32 @@ public int writeIntoBuffer(ByteBuffer buffer) {
         int totalBytesWritten = 0;
 
         // Write the protocol type.
-        buffer.put((byte) protocol.ordinal());
-        totalBytesWritten += 1; // size of byte
+        if (identifierType == NanoTDFType.IdentifierType.NONE) {
+            buffer.put((byte) protocol.ordinal());
+            totalBytesWritten += 1; // size of byte
+        } else {
+            buffer.put((byte) (identifierType.ordinal() << 4 | protocol.ordinal()));
+            totalBytesWritten += 1;
+        }
 
-        // Write the url body length;
+        // Write the url body length
         buffer.put((byte)bodyLength);
         totalBytesWritten += 1;
 
-        // Write the url body;
+        // Write the url body
         buffer.put(body);
         totalBytesWritten += body.length;
 
+        // Write the identifier
+        if (identifierType != NanoTDFType.IdentifierType.NONE) {
+            buffer.put(identifier);
```

Comment:
You may need to pad the buffer here or assert (throw IllegalArg) if identifier is. the wrong length

Reply:
Note we have this setter (without getter)
```
public void setProtocol()
```

Reply:
done added padded and tests